### PR TITLE
seatd: rename from libseat; build binary too; enable embedded seatd in library

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

"seatd" is the name of the upstream project, so it's the proper name for the package containing both the binary and the library.

The builtin seatd backend is the new mechanism for wlroots-based compositors to be able to run as root, with no logind or seatd running.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by Emantor — APPROVED** [review]
I mostly added this as `libseat` since NixOS integrates sytemd by default, but for other distributions using nix packages this makes sense.

File: pkgs/applications/misc/seatd/default.nix [hunk 1]
```diff
@@ -18,13 +19,13 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-JwlJLHkRgSRqfQEhXbzuFTmhxfbwKVdLICPbTDbC9M0=";
   };
 
-  nativeBuildInputs = [ meson ninja pkg-config ];
+  outputs = [ "bin" "out" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/seatd/default.nix:22]: You are sure we do not want dev here, too?

Comment (alyssais) [reply pkgs/applications/misc/seatd/default.nix]: Yeah makes sense.  I've added it, and a man output.

**Review by primeos — APPROVED** [review]
Diff LGTM and `nixosTests.sway` still succeeds.

**Pull request merged**